clustered/common/src/main/java/org/ehcache/clustered/common/store/Element.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.common.store;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Building blocks of {@link Chain}
+ */
+public interface Element {
```

Comment:
Couldn't facilitation of such optimizations be added to the abstraction (or as an additional abstraction) at the point when we'd be adding such optimizations?

Let's not over-complicate things for now - keep the scope constrained.